atoma-confidential/src/service.rs

```diff
@@ -131,16 +130,18 @@ impl AtomaConfidentialCompute {
     ) -> Result<Self> {
         let key_manager = X25519KeyPairManager::new()?;
         let mut is_cc_supported = false;
-        for index in &config.device_indices {
-            is_cc_supported &= check_confidential_compute_status(u32::from(*index))?;
+        // NOTE: If the NVML library is not available, we return 0 devices
+        let num_devices = num_devices().unwrap_or(0);
+        for index in 0..num_devices {
+            is_cc_supported &= check_confidential_compute_status(index)?;
```

Comment:
this is actually always false, the initial value should be true (but probably only in cases when the num_devices is at least 1

Reply:
Oh nice catch !